karapace/schema_models.py

```diff
@@ -79,6 +79,8 @@ def __init__(self, schema_type: SchemaType, schema_str: str):
         self.schema_str = schema_str
         self.max_id: Optional[int] = None
 
+        self._str_cached: Optional[str] = None
```

Comment:
This doubles the memory usage? Does that matter?

Reply:
There are other memory usage and this does not apply to all schema types, so it does not double all memory usage. But definately this needs followed, and if there comes unifying changes for normalization of schemas, it reduces memory usage of original schema definitions here.